v3/integrations/nrsecurityagent/nrsecurityagent.go

```diff
@@ -33,6 +33,17 @@ func defaultSecurityConfig() SecurityConfig {
 	return cfg
 }
 
+// To completely disable security set NEW_RELIC_SECURITY_AGENT_ENABLED env to false.
+// If env is set to false,the security module is not loaded
```

Comment:
This seems to be breaking with how configuration works for the agent and the nrsecurityagent integration, in that this keys exclusively on an environment variable while the rest of the configuration is done via environment variable, code, and/or config file. Instead of checking separately for the environment variable here, this code should check with the integration configuration.

Reply:
To disable the security agent we have two config flags.

```
c.Security.Enabled           // This is used to disable security agent functionality (hooks are applied but no event will be generated)
c.Security.Agent.Enabled // This is used to completely disable security. (security module and hooks are not loaded)

```
In csec-go-agent we apply hooks at the time of package initializing. At that time we don't have a config object that's why to completely disable security set `NEW_RELIC_SECURITY_AGENT_ENABLED` env to false.